bin/test-lxd-storage-vm

```diff
@@ -431,8 +431,19 @@
         lxc restore v1 snap1
         newUUID=$(lxc config get v1 volatile.uuid)
 
-        if [ "${initialUUID}" = "${newUUID}" ]; then
-                echo "==> UUID of the VM instance should be different after restoring its snapshot"
+        if [ "${initialUUID}" != "${newUUID}" ]; then
+                echo "==> UUID of the VM instance should remain the same after restoring its snapshot"
+                false
+        fi
+
+        # Check that the QEMU process assocciated with this VM contains the correct VM Generation ID.
+        # This is used by the QEMU migration code to ensure that the VM is not migrated to a host
+        # that has a different VM Generation ID.
+        vmGenID=$(lxc config get v1 volatile.uuid.vmgenid)
+        vmPID=$(lxc info v1 | grep "PID:" | awk '{print $2}')
+        qemuGenID=$(ps -o args= -p "${vmPID}" | grep -o "vmgenid,guid=[^,]*" | cut -d "=" -f2 | cut -d " " -f1)
```

Comment:
`ps -q "${vmPID}" -ho cmd | sed -n 's/.*vmgenid,guid=\([^,]\+\).*/\1/p'` should work (TM)

Reply:
Can we use pgrep to appease shellcheck

Reply:
`pgrep` can't look at a single PID AFAIK. If we really want to avoid `ps` here, we can probably go with:

```
sed -n 's/.*vmgenid,guid=\([^,]\+\).*/\1/p' /proc/"${vmPID}"/cmdline
```

Reply:
Yes that would be nice.